lib/needle.js

```diff
@@ -574,7 +574,7 @@ Needle.prototype.send_request = function(count, method, uri, config, post_data,
 
     var pipeline      = [],
         mime          = parse_content_type(headers['content-type']),
-        text_response = mime.type && mime.type.indexOf('text/') != -1;
+        text_response = mime.type && (mime.type.indexOf('text/') != -1 || ["/json","+json"].indexOf(mime.type.slice(-5)) != -1 || ["/xml","+xml"].indexOf(mime.type.slice(-4)) != -1);
```

Comment:
Any chance we could simplify this a bit? Maybe use a regex, eg. `mime.type.match(/text|json|xml/)`

Reply:
i was deliberately avoiding the use of regular expressions for performance reasons, but feel free to use `mime.type.test(/(\/|\+)(xml|json)$/)` if elegance is your priority.